### PR TITLE
feat: new [components `<Head>` and `<Config>`](https://vike.dev/useConfig#config-head)

### PR DESCRIPTION
**Pull request by brillout**
Is it possible to implement [the `<Config>` and `<Head>` components](https://vike.dev/useConfig#config-head) for Vue?

How can the children be "accessed" so that we can pass it to `useConfig({ Head })`?

Directly using `useConfig()` does work, see [this example](https://github.com/vikejs/vike-vue/blob/c1f988837f8b90ebe2e43b258137916e30b8f7be/examples/full/pages/images/Image.vue#L15-L32). So the only thing missing is to be able to "access" the children of `<Head>`. Vue seems to work in terms of slots, not in terms of children; I don't know how we can "pass the children" with Vue.

Contribution welcome :green_heart:

### CODE REVIEW

**Review by pdanpdan — COMMENTED** [review]
I should have put a comment. The regex must work as if 's' modifier is set for '.', so instead of writing a positive it's easier to write a negative for something that it's not there.

**Comment by brillout** [comment]
Yea I've realized, let me add support for regex flags to @brillout/part-regex.

File: examples/full/.testRun.ts [hunk 1]
```diff
@@ -226,3 +277,12 @@ function getAssetUrl(fileName: string) {
   assert(r.length === 0)
   return partRegex`/assets/static/${fileBaseName}.${/[a-zA-Z0-9_-]+/}.${fileExt}`
 }
+
+function countMatches(haystack, needleRe) {
+  const re = new RegExp(needleRe, 'g')
```
Comment (pdanpdan) [inline examples/full/.testRun.ts:282]: If you add support for flags then this can be also simplified.

**Comment by brillout** [comment]
Merged :rocket: 

Thank you for the nice collaboration 🤝

I've added our work to https://vike.dev/releases/2024-08#improved-head-management.

**Pull request merged**